src/Provider/YahooApi.php

```diff
@@ -1,38 +0,0 @@
-<?php
-namespace CurrencyConverter\Provider;
-
-class YahooApi implements ProviderInterface
-{
-    /**
-     * Url where Curl request is made
-     *
-     * @var strig
-     */
-    const API_URL = 'http://download.finance.yahoo.com/d/quotes.csv?s=[fromCurrency][toCurrency]=X&f=nl1d1t1';
-
-    /**
-     * {@inheritDoc}
-     */
-    public function getRate($fromCurrency, $toCurrency)
-    {
-        $fromCurrency = urlencode($fromCurrency);
-        $toCurrency = urlencode($toCurrency);
-
-        $url = str_replace(
-            ['[fromCurrency]', '[toCurrency]'],
-            [$fromCurrency, $toCurrency],
-            static::API_URL
-        );
```

Comment:
`sprintf` is probably more performant than str_replace especially in predictable string replace scenarios. I don't have the stats. But, that's my guess.